src/data-workspace/inputs/generic-input.js

```diff
@@ -43,23 +43,51 @@ export const GenericInput = ({
         )
     }
 
+    const formatValue = (value) => {
+        let formattedValue = value
+
+        if (value && valueType === VALUE_TYPES.NUMBER) {
```

Comment:
also relevant for UNIT_INTERVAL and PERCENTAGE valueTypes

Reply:
Thanks thats useful - I was wondering if over value types should be considered here too  

Reply:
I think this applies to all values types in either `CAN_HAVE_LIMITS_TYPES` or `NUMBER_TYPES` (see [`value-types.js`](https://github.com/dhis2/data-entry-app/blob/TECH-1246/handling-leading-zeros-and-decimal-points/src/shared/value-types.js)).

Reply:
Awesome :-) thanks @Mohammer5 I'll make that change